### DISCUSSION TITLE
Wrap code in <details> when outputting to GFM

### DISCUSSION THREAD

**Discussion by mavam** [quarto-dev/quarto-cli#2278] [Q&A] [answered]
I'm outputting my qmd in GFM, which puts code and output adjacent to each other. I would like to wrap *only the code* in a `<details>` section:

I tried this:

    <details><summary>Code</summary>

    ```{r}
    data |> ggplot(..)
    ```

    </details>

But this also puts the output (in this case, a plot) into the details section. I couldn't find an option for that in the [GFM docs](https://quarto.org/docs/reference/formats/markdown/gfm.html).

My hunch is that I need to write a filter, ~but wrangling the AST in Lua is a different ballpark.~  EDIT: I managed to write a filter, but it feels pretty hacky:

```lua
function CodeBlock(code_block)
    if code_block.attributes["code-fold"] == "true" then
        local open = "<details><summary>Code </summary>\n"
        local close = "\n</details>"
        local open_block = pandoc.RawBlock("html", open)
        local close_block = pandoc.RawBlock("html", close)
        return {open_block, code_block, close_block}
    end
    return code_block
end
```

Is that the right approach?

(NB: I'm aware that HTML output does this. But I need to output to GFM where there is no collapse option.)

**Comment by mcanouil** [comment, marked as answer]
In my opinion, "the right approach" would be for Quarto to handle this natively.

Reusable chunk is another way (probably the easiest) for R code cells.

````
---
format: gfm
---

<details><summary>Code</summary>

```{r}
#| eval: false
#| label: my-ggplot
1 + 1
```

</details>

```{r}
#| echo: false
#| label: my-ggplot
```
````

To note and as pointed out, `code-fold: true` will do the same thing as above but for html-based formats.
Possibly a feature request (#2279) to extend that option to other formats such as `gfm`.

````
---
format: 
  html: default
  gfm: default
code-fold: true
---

```{r}
1 + 1
```
````

**Reply by mavam** [reply]
Continuing the discussion in #2279.
